src/RevSysCalls.cc

```diff
@@ -620,6 +620,43 @@ EcallStatus RevCore::ECALL_openat() {
   return EcallLoadAndParseString( pathname, action );
 }
 
+// 1024, int rev_open(const char *filename, int flags)
+EcallStatus RevCore::ECALL_open() {
+  auto& EcallState = Harts.at( HartToExecID )->GetEcallState();
+  if( EcallState.bytesRead == 0 ) {
+    output->verbose(
+      CALL_INFO, 2, 0, "ECALL: open called by thread %" PRIu32 " on hart %" PRIu32 "\n", ActiveThreadID, HartToExecID
+    );
+  }
+  auto pathname = RegFile->GetX<uint64_t>( RevReg::a0 );
+
+  auto flags = RegFile->GetX<int>(RevReg::a1);
+
+  /* Read the filename from memory one character at a time until we find '\0' */
+
+  auto action   = [&] {
+    // Do the open on the host
+    std::string const full_path =
+      std::filesystem::current_path().append(EcallState.string).string();
+    auto fd  = open( full_path.c_str(), flags);
+
+    if(fd != -1) {
+      // Add the file descriptor to this thread
+      Harts.at( HartToExecID )->Thread->AddFD( fd );
+    }
+    else {
+      output->verbose(
+        CALL_INFO, 2, 0, "ECALL: open called by thread %" PRIu32 " on hart %" PRIu32 "; file descriptor invalid (fd == -1)\n", ActiveThreadID, HartToExecID
```

Comment:
I would use:

`"ECALL: open called by thread %" PRIu32 " on hart %" PRIu32 " returned -1:%m\n"`

 Nothing about "file descriptor invalid". `%m` will print the cause of the error, such as `No such file or directory`.